tests/src/test/java/io/streamnative/pulsar/handlers/kop/KafkaRequestTypeTest.java

```diff
@@ -214,6 +215,56 @@ public void testKafkaProducePulsarConsume(int partitionNumber, boolean isBatch)
         assertNull(msg);
     }
 
+    @Test(timeOut = 20000, dataProvider = "partitionsAndBatch")
+    public void testKafkaProducePulsarMetrics(int partitionNumber, boolean isBatch) throws Exception {
+        String kafkaTopicName = "kopKafkaProducePulsarMetrics" + partitionNumber;
+        String pulsarTopicName = "persistent://public/default/" + kafkaTopicName;
+        String key1 = "header_key1_";
+        String key2 = "header_key2_";
+        String value1 = "header_value1_";
+        String value2 = "header_value2_";
+
+        // create partitioned topic.
+        admin.topics().createPartitionedTopic(kafkaTopicName, partitionNumber);
+
+        // 1. produce message with Kafka producer.
+        @Cleanup
+        KProducer kProducer = new KProducer(kafkaTopicName, false, getKafkaBrokerPort());
+
+        int totalMsgs = 10;
+
+        String messageStrPrefix = "Message_Kop_KafkaProducePulsarConsume_"  + partitionNumber + "_";
+
+        for (int i = 0; i < totalMsgs; i++) {
+            String messageStr = messageStrPrefix + i;
+            ProducerRecord record = new ProducerRecord<>(
+                    kafkaTopicName,
+                    i,
+                    messageStr);
+            record.headers()
+                    .add(key1 + i, (value1 + i).getBytes(UTF_8))
+                    .add(key2 + i, (value2 + i).getBytes(UTF_8));
```

Comment:
Could you explain why add headers here? I didn't find anywhere to check the headers.

Reply:
I planed to compute the byte size of the record with headers, it can remove now.